bpfman/src/serve.rs

```diff
@@ -162,3 +181,38 @@ async fn serve_unix(
         info!("Shutdown Unix Handler {}", path);
     }))
 }
+
+fn systemd_unix_stream(_path: String) -> anyhow::Result<UnixListenerStream> {
+    let listen_fds = libsystemd::activation::receive_descriptors(true)?;
+    if listen_fds.len() == 1 {
+        if let Some(fd) = listen_fds.first() {
+            if !fd.is_unix() {
+                return Err(anyhow!("Wrong Socket"));
+            }
+            let std_listener =
+                unsafe { std::os::unix::net::UnixListener::from_raw_fd(fd.clone().into_raw_fd()) };
+            std_listener.set_nonblocking(true)?;
+            let tokio_listener = UnixListener::from_std(std_listener)?;
+            info!("Using a Unix socket from systemd");
+            return Ok(UnixListenerStream::new(tokio_listener));
+        }
+    }
+
+    Err(anyhow!("Unable to retrieve fd from systemd"))
+}
+
+async fn std_unix_stream(path: String) -> anyhow::Result<UnixListenerStream> {
+    // Listen on Unix socket
+    if Path::new(&path).exists() {
+        // Attempt to remove the socket, since bind fails if it exists
+        remove_file(&path)?;
+    }
+
+    let uds = UnixListener::bind(&path)?;
+    let stream = UnixListenerStream::new(uds);
+    // Always set the file permissions of our listening socket.
+    set_file_permissions(&path.clone(), SOCK_MODE).await;
+
+    info!("Using default Unix socket");
+    Ok(stream)
```

Comment:
```suggestion
    // Always set the file permissions of our listening socket.
    set_file_permissions(&path.clone(), SOCK_MODE).await;

    info!("Using default Unix socket");
    Ok(UnixListenerStream::new(uds));
```

Can you just do this?